node/src/components/contract_runtime/utils.rs

```diff
@@ -209,14 +210,14 @@ pub(super) async fn exec_or_requeue<REv>(
                 let era_score = { Ratio::new(utilization, block_count).to_integer() };
 
                 let new_gas_price = if era_score >= go_up {
-                    let new_gas_price = current_gas_price + 1;
+                    let new_gas_price = current_gas_price.saturating_add(1);
```

Comment:
not blocking over this, but next week id like to better understand why we're calculating this here instead of getting it from the block itself